vmdk_plugin/plugin.go

```diff
@@ -208,12 +209,55 @@ func (d *vmdkDriver) unmountVolume(name string) error {
 // (until Mount is called).
 // Name and driver specific options passed through to the ESX host
 func (d *vmdkDriver) Create(r volume.Request) volume.Response {
-	err := d.ops.Create(r.Name, r.Options)
+	// Use default fstype if not specified
+	if _, ok := r.Options["fstype"]; ok == false {
+		r.Options["fstype"] = fs.FstypeDefault
+	}
+
+	// Verify the existence of filesystem tools
+	mkfscmd := "/sbin/mkfs." + r.Options["fstype"]
+	_, err := os.Lstat(mkfscmd)
+	if err != nil {
+		log.WithFields(log.Fields{"name": r.Name, "error": err}).Error("Not found ")
+		return volume.Response{Err: err.Error() + "\nSupported filesystems found: " + fs.MkfsLookup()}
+	}
+
+	err = d.ops.Create(r.Name, r.Options)
 	if err != nil {
 		log.WithFields(log.Fields{"name": r.Name, "error": err}).Error("Create volume failed ")
 		return volume.Response{Err: err.Error()}
 	}
-	log.WithFields(log.Fields{"name": r.Name}).Info("Volume created ")
+
+	// Handle filesystem creation
+	log.WithFields(log.Fields{"name": r.Name, "fstype": r.Options["fstype"]}).Info("Attaching volume and creating filesystem ")
```

Comment:
nit: this is much longer that 80 chars. makes it hard to use dual pages in an editor on  a laptop....